Install needed packages (on a Debian-based system):
apt-get install git make
apt-get install --no-install-recommends asciidoc

Generating git docbook documentation:
- Clone git
- apply patch
- change (system/global) asciidoc.conf
  - comment:
    lsquo=&#8216;
    rsquo=&#8217;
    ldquo=&#8220;
    rdquo=&#8221;
    (^-- )=&#8212;&#8201;
    (\n-- )|( -- )|( --\n)=&#8201;&#8212;&#8201;
    (\w)--(\w)=\1&#8212;\2
    (\w)'(\w)=\1&#8217;\2
- perl -0pi -e 's/link:user-manual.html#(\S+)\[(.+?)\]/xref:user-manual_\1\[\2]/gms' *.adoc
- perl -pi -e 's/link:user-manual.html\[(.+?)\]/xref:git_user-manual[\1]/g' *.adoc
- make xml
- take *.xml files
- Update git-doc-version.xml

diff --git a/Documentation/Makefile b/Documentation/Makefile
index b109d25..e93261d 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -250,6 +250,8 @@ all:: html man
 
 html: $(DOC_HTML)
 
+xml: cmd-list.sh $(MAN_XML) user-manual.xml
+
 man: man1 man5 man7
 man1: $(DOC_MAN1)
 man5: $(DOC_MAN5)
@@ -353,7 +355,7 @@ docinfo.html: docinfo-html.in
 	$(QUIET_GEN)$(RM) $@ && cat $< >$@
 
 $(MAN_HTML): %.html : %.adoc $(ASCIIDOC_DEPS)
-	$(QUIET_ASCIIDOC)$(TXT_TO_HTML) -d manpage -o $@ $<
+	$(QUIET_ASCIIDOC)$(TXT_TO_HTML) -d article -o $@ $<
 
 $(OBSOLETE_HTML): %.html : %.adoco $(ASCIIDOC_DEPS)
 	$(QUIET_ASCIIDOC)$(TXT_TO_HTML) -o $@ $<
@@ -369,10 +371,10 @@ manpage-cmd = $(QUIET_XMLTO)$(XMLTO) -m $(MANPAGE_XSL) $(XMLTO_EXTRA) man $<
 	$(manpage-cmd)
 
 %.xml : %.adoc $(ASCIIDOC_DEPS)
-	$(QUIET_ASCIIDOC)$(TXT_TO_XML) -d manpage -o $@ $<
+	$(QUIET_ASCIIDOC)$(TXT_TO_XML) -d article -o $@ $<
 
 user-manual.xml: user-manual.adoc $(ASCIIDOC_DEPS)
-	$(QUIET_ASCIIDOC)$(TXT_TO_XML) -d book -o $@ $<
+	$(QUIET_ASCIIDOC)$(TXT_TO_XML) -d article -o $@ $<
 
 technical/api-index.adoc: technical/api-index-skel.adoc \
 	technical/api-index.sh $(patsubst %,%.adoc,$(API_DOCS))
diff --git a/Documentation/asciidoc.conf.in b/Documentation/asciidoc.conf.in
index 9d91393..8049f08 100644
--- a/Documentation/asciidoc.conf.in
+++ b/Documentation/asciidoc.conf.in
@@ -23,7 +23,6 @@ backtick=&#96;
 litdd=&#45;&#45;
 manmanual=Git Manual
 mansource=Git @GIT_VERSION@
-revdate=@GIT_DATE@
 
 ifdef::doctype-book[]
 [titles]
@@ -43,16 +42,178 @@ ifdef::doctype-book[]
 endif::doctype-book[]
 
 [literal-inlinemacro]
-{eval:re.sub(r'(&lt;[-a-zA-Z0-9.]+&gt;)', r'<emphasis>\1</emphasis>', re.sub(r'([\[\s|()>]|^|\]|&gt;)(\.?([-a-zA-Z0-9:+=~@\\\*\/_^\$]+\.?)+|,)',r'\1<literal>\2</literal>', re.sub(r'(\.\.\.?)([^\]$.])', r'<literal>\1</literal>\2', macros.passthroughs[int(attrs['passtext'][1:-1])] if attrs['passtext'][1:-1].isnumeric() else attrs['passtext'][1:-1])))}
+{eval:re.sub(r'(&lt;[-a-zA-Z0-9.]+&gt;)', r'<emphasis>\1</emphasis>', re.sub(r'([\[\s|()>]|^|\]|&gt;)(\.?([-a-zA-Z0-9:+=~@,\/_^\$]+\.?)+)',r'\1<literal>\2</literal>', re.sub(r'(\.\.\.?)([^\]$.])', r'<literal>\1</literal>\2', macros.passthroughs[int(attrs['passtext'][1:-1])] if attrs['passtext'][1:-1].isnumeric() else attrs['passtext'][1:-1])))}
 
 endif::backend-docbook[]
 
+ifdef::backend-docbook[]
+[header-declarations]
+<?xml version="1.0" encoding="{encoding}"?>
+<!DOCTYPE {doctype-article?sect2}{doctype-book?book}{doctype-manpage?refentry} SYSTEM "../../../dtd/dblite.dtd">
+
+ifdef::doctype-article[]
+[header]
+template::[header-declarations]
+
+<sect2 lang="{lang=en}" id="{doctitle}">
+template::[docinfo]
+<indexterm>
+<primary>{doctitle}</primary>
+</indexterm>
+
+[footer]
+</sect2>
+
+[callout-inlinemacro]
+# Callout.
+<co id="{docname}_{coid}"/>
+
+[listtags-callout]
+list=<calloutlist{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>{title?<title>{title}</title>}|</calloutlist>
+item=<callout arearefs="{docname}_{coids}">|</callout>
+text=<para>|</para>
+
+# [[id,text]]
+[anchor2-inlinemacro]
+<anchor id="{doctitle}_{1}" xreflabel="{2=[{1}]}"/>
+# [[[id]]]
+[anchor3-inlinemacro]
+<anchor id="{doctitle}_{1}" xreflabel="[{1}]"/>[{1}]
+# <<id,text>>
+[xref2-inlinemacro]
+<link linkend="{doctitle}_{1}">{2}</link>
+{2%}<xref linkend="{doctitle}_{1}"/>
+
+[appendix]
+<sect3{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</sect3>
+
+[linkgit-inlinemacro]
+<xref linkend="{target}{0?({0})}" />
+
+[link-inlinemacro]
+{0%{target}}
+<ulink url="https://www.kernel.org/pub/software/scm/git/docs/{target}"><citetitle>{0}</citetitle></ulink>
+
+[sect1]
+<simplesect{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</simplesect>
+
+[sect2]
+<section{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[sect3]
+<section{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[sect4]
+<section{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[literal-inlinemacro]
+# Inline literal.
+<emphasis>{passtext}</emphasis>
+
+# Paragraph substitution.
+[paragraph]
+<formalpara{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}><title>{title}</title><para>
+{title%}<simpara{id? id="{doctitle}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+|
+{title%}</simpara>
+{title#}</para></formalpara>
+{empty}
+
+[quotes]
+ifdef::no-inline-literal[]
+`=literal
+endif::no-inline-literal[]
+
+[preface]
+<section{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[paradef-default]
+synopsis-style=template="verseparagraph",filter="sed 's!\\([\\[ |()]\\|^\\|\\]\\|&gt;\\)\\([-=a-zA-Z0-9:+@,\\/_^\\$.]\\+\\|&#8230;\\)!\\1<code>\\2</code>!g;s!&lt;[-a-zA-Z0-9.]\\+&gt;!<emphasis>\\0</emphasis>!g'"
+
+ifeval::["{doctitle}" == "Git User Manual"]
+[header]
+template::[header-declarations]
+
+<sect2 lang="{lang=en}" id="git_user-manual">
+template::[docinfo]
+
+# [[id,text]]
+[anchor2-inlinemacro]
+<anchor id="{docname}_{1}" xreflabel="{2=[{1}]}"/>
+# [[[id]]]
+[anchor3-inlinemacro]
+<anchor id="{docname}_{1}" xreflabel="[{1}]"/>[{1}]
+# <<id,text>>
+[xref2-inlinemacro]
+<link linkend="{docname}_{1}">{2}</link>
+{2%}<xref linkend="{docname}_{1}"/>
+
+[sect1]
+<section{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[sect2]
+<section{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[sect3]
+<section{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[sect4]
+<section{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</section>
+
+[paragraph]
+<formalpara{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}><title>{title}</title><para>
+{title%}<simpara{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+|
+{title%}</simpara>
+{title#}</para></formalpara>
+{empty}
+
+[appendix]
+<sect3{id? id="{docname}_{id}"}{role? role="{role}"}{reftext? xreflabel="{reftext}"}>
+<title>{title}</title>
+|
+</sect3>
+endif::[]
+
+endif::doctype-article[]
+endif::backend-docbook[]
+
 ifdef::backend-docbook[]
 ifdef::doctype-manpage[]
 # The following two small workarounds insert a simple paragraph after screen
 [listingblock]
 <example><title>{title}</title>
-<literallayout class="monospaced">
+<literallayout>
 |
 </literallayout><simpara></simpara>
 {title#}</example>
@@ -75,18 +236,18 @@ git-relative-html-prefix=
 <a href="{git-relative-html-prefix}{target}.html">{target}{0?({0})}</a>
 
 [literal-inlinemacro]
-{eval:re.sub(r'(&lt;[-a-zA-Z0-9.]+&gt;)', r'<em>\1</em>', re.sub(r'([\[\s|()>]|^|\]|&gt;)(\.?([-a-zA-Z0-9:+=~@,\\\*\/_^\$]+\.?)+)',r'\1<code>\2</code>', re.sub(r'(\.\.\.?)([^\]$.])', r'<code>\1</code>\2', macros.passthroughs[int(attrs['passtext'][1:-1])] if attrs['passtext'][1:-1].isnumeric() else attrs['passtext'][1:-1])))}
+{eval:re.sub(r'(&lt;[-a-zA-Z0-9.]+&gt;)', r'<em>\1</em>', re.sub(r'([\[\s|()>]|^|\]|&gt;)(\.?([-a-zA-Z0-9:+=~@,\/_^\$]+\.?)+)',r'\1<code>\2</code>', re.sub(r'(\.\.\.?)([^\]$.])', r'<code>\1</code>\2', macros.passthroughs[int(attrs['passtext'][1:-1])] if attrs['passtext'][1:-1].isnumeric() else attrs['passtext'][1:-1])))}
 
 endif::backend-xhtml11[]
 
 ifdef::backend-docbook[]
 ifdef::doctype-manpage[]
 [paradef-default]
-synopsis-style=template="verseparagraph",filter="sed 's!&#8230;\\(\\]\\|$\\)!<phrase>\\0</phrase>!g;s!\\([\\[ |()]\\|^\\|\\]\\|&gt;\\)\\([-=a-zA-Z0-9:+@,\\/_^\\$.\\\\\\*]\\+\\|&#8230;\\)!\\1<literal>\\2</literal>!g;s!&lt;[-a-zA-Z0-9.]\\+&gt;!<emphasis>\\0</emphasis>!g'"
+synopsis-style=template="verseparagraph",filter="sed 's!&#8230;\\(\\]\\|$\\)!<phrase>\\0</phrase>!g;s!\\([\\[ |()]\\|^\\|\\]\\|&gt;\\)\\([-=a-zA-Z0-9:+@,\\/_^\\$.]\\+\\|&#8230;\\)!\\1<literal>\\2</literal>!g;s!&lt;[-a-zA-Z0-9.]\\+&gt;!<emphasis>\\0</emphasis>!g'"
 endif::doctype-manpage[]
 endif::backend-docbook[]
 
 ifdef::backend-xhtml11[]
 [paradef-default]
-synopsis-style=template="verseparagraph",filter="sed 's!&#8230;\\(\\]\\|$\\)!<span>\\0</span>!g;s!\\([\\[ |()]\\|^\\|\\]\\|&gt;\\)\\([-=a-zA-Z0-9:+@,\\/_^\\$.\\\\\\*]\\+\\|&#8230;\\)!\\1<code>\\2</code>!g;s!&lt;[-a-zA-Z0-9.]\\+&gt;!<em>\\0</em>!g'"
+synopsis-style=template="verseparagraph",filter="sed 's!&#8230;\\(\\]\\|$\\)!<span>\\0</span>!g;s!\\([\\[ |()]\\|^\\|\\]\\|&gt;\\)\\([-=a-zA-Z0-9:+@,\\/_^\\$.]\\+\\|&#8230;\\)!\\1<code>\\2</code>!g;s!&lt;[-a-zA-Z0-9.]\\+&gt;!<em>\\0</em>!g'"
 endif::backend-xhtml11[]
